src/ucp/core/ucp_worker.c

```diff
@@ -3185,18 +3197,20 @@ void ucp_worker_keepalive_remove_ep(ucp_ep_h ep)
 {
     ucp_worker_h worker = ep->worker;
 
-    ucs_assert(!(ep->flags & UCP_EP_FLAG_INTERNAL));
-
-    if (!ucp_worker_keepalive_is_enabled(worker)) {
-        ucs_assert(worker->keepalive.iter == &worker->all_eps);
+    if ((ep->flags & UCP_EP_FLAG_INTERNAL) ||
```

Comment:
do we really have to check UCP_EP_FLAG_INTERNAL, given that also check ucp_ep_config(ep)->key.keepalive_lane == UCP_NULL_LANE?
could replace by assert